### PR TITLE
feat: resource recently updated

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/3857

### Description:

homepage:

CleanShot 2023-01-30 at 13 12 43@2x

homepage empty state:

CleanShot 2023-01-30 at 13 15 02@2x

resource empty states:

all dash / without space:
CleanShot 2023-01-30 at 13 09 58@2x
all dash / with space
CleanShot 2023-01-30 at 13 11 01@2x
all saved charts:
CleanShot 2023-01-30 at 13 10 04@2x

spaces page:
CleanShot 2023-01-30 at 13 16 25@2x

space page:
CleanShot 2023-01-30 at 13 10 23@2x

### CODE REVIEW

File: packages/frontend/src/components/Home/RecentlyUpdatedPanel/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    const recentItems = useMemo(() => {
+        return [...dashboards, ...savedCharts]
+            .sort((a, b) => {
+                return (
+                    new Date(b.updatedAt).getTime() -
+                    new Date(a.updatedAt).getTime()
+                );
+            })
+            .slice(0, 10);
```
Comment (IrakliJani) [inline packages/frontend/src/components/Home/RecentlyUpdatedPanel/index.tsx:30]: showing the most recent 10 items after sorting dash and charts

File: packages/frontend/src/components/Home/RecentlyUpdatedPanel/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                        <ResourceEmptyStateHeader>
+                            Feels a little bit empty over here...
+                        </ResourceEmptyStateHeader>
+
+                        <ResourceEmptyStateText>
+                            get started by creating some charts
+                        </ResourceEmptyStateText>
+                    </ResourceEmptyStateHeaderWrapper>
+
+                    <Button icon="plus" intent="primary">
```
Comment (ZeRego) [inline packages/frontend/src/components/Home/RecentlyUpdatedPanel/index.tsx:63]: missing on click behaviour

File: packages/frontend/src/pages/SavedDashboards.tsx [hunk 3]
```diff
@@ -143,18 +145,28 @@ const SavedDashboards = () => {
...
+
+                            <ResourceEmptyStateHeader>
+                                No dashboards added yet
+                            </ResourceEmptyStateHeader>
+
+                            {!isDemo &&
+                                !hasNoSpaces &&
+                                userCanManageDashboards && (
+                                    <Button
+                                        text="Create chart"
```
Comment (ZeRego) [inline packages/frontend/src/pages/SavedDashboards.tsx:162]:
```suggestion
                                        text="Create dashboard"
```

File: packages/frontend/src/pages/Home.tsx [hunk 4]
```diff
@@ -60,11 +62,14 @@ const Home: FC = () => {
...
                     <LandingPanel
-                        hasSavedChart={!!savedChartStatus.data}
                         userName={user.data?.firstName}
                         projectUuid={project.data.projectUuid}
                     />
                 )}
+
+                <SpaceBrowser projectUuid={project.data.projectUuid} />
+
+                <RecentlyUpdatedPanel projectUuid={project.data.projectUuid} />
```
Comment (ZeRego) [inline packages/frontend/src/pages/Home.tsx:72]: Do we want to show these with the onboarding panel?

File: packages/frontend/src/components/common/ResourceList/ResourceTable.tsx [hunk 5]
```diff
@@ -99,6 +99,22 @@ const ResourceTable: FC<ResourceTableProps> = ({
...
+                return `/projects/${projectUuid}/dashboards/${resource.uuid}/view`;
+            case 'chart':
+                return `/projects/${projectUuid}/saved/${resource.uuid}`;
+            default:
+                return assertUnreachable(
+                    resourceType,
+                    `Can't get URL for ${resourceType}`,
+                );
+        }
+    };
```
Comment (ZeRego) [inline packages/frontend/src/components/common/ResourceList/ResourceTable.tsx:116]: We could move this outside of the component, so we don't need to add it to the hook dependencies

File: packages/frontend/src/pages/Home.tsx (hunk 4 above)
Comment (IrakliJani) [reply packages/frontend/src/pages/Home.tsx]: good catch

File: packages/frontend/src/components/Home/RecentlyUpdatedPanel/index.tsx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+                            get started by creating some charts
+                        </ResourceEmptyStateText>
+                    </ResourceEmptyStateHeaderWrapper>
+
+                    <Button
+                        icon="plus"
+                        intent="primary"
+                        onClick={handleCreateChart}
+                    >
+                        Create chart
```
Comment (IrakliJani) [inline packages/frontend/src/components/Home/RecentlyUpdatedPanel/index.tsx:74]: missing permission check

File: packages/frontend/src/components/Explorer/SpaceBrowser/index.tsx [hunk 7]
```diff
@@ -56,15 +60,24 @@ const SpaceBrowser: FC<{ projectUuid: string }> = ({ projectUuid }) => {
...
+                    </ResourceEmptyStateHeader>
+
+                    <Button
+                        text="Create space"
+                        icon="plus"
+                        intent="primary"
+                        onClick={
+                            !isDemo && userCanManageSpace
+                                ? handleCreateSpace
+                                : undefined
```
Comment (IrakliJani) [inline packages/frontend/src/components/Explorer/SpaceBrowser/index.tsx:77]: missing permission check

**Review by ZeRego — APPROVED** [review]

**Pull request merged**